sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/CosmosDiagnosticsE2ETest.java

```diff
@@ -197,17 +219,28 @@ public void delayedSampling() {
         meterRegistry.clear();
         meterRegistry.close();
 
+        String id = UUID.randomUUID().toString();
+        ObjectNode newItem = getDocumentDefinition(id);
+        container.createItem(newItem);
+
         // change sample rate to 25%
         clientTelemetryCfg.sampleDiagnostics(0.25);
-        executeTestCase(container);
+        executeDocumentOperation(container, operationType, id, newItem);
 
+        int loggedMessageSizeBefore = capturingLogger.getLoggedMessages().size();
```

Comment:
Just curious - isn't there a non-zero chance of `loggedMessageSizeBefore` to have 1 since sampling rate is 25%?

Reply:
yea, that is why the test is just comparing the before & after for sampleRate = 0 or sampleRate=1

Reply:
Resolved offline.